### PR TITLE
Spline tool polish

### PR DESCRIPTION
**Pull request by Fredaly**
bugs fixes and add some polish features

### CODE REVIEW

**Comment by Fredaly** [comment]
Added a Units enum from Niagara Plugin in BP_SplineMarker
Since Niagara Plugin is on every unreal project it is fine to keep it ?

File: Source/SplineFollow/Public/Components/SFSplineComponent.h [hunk 1]
```diff
@@ -37,6 +37,9 @@ class SPLINEFOLLOW_API USFSplineComponent final : public USplineComponent
     void Serialize( FArchive & archive );
 
 #if WITH_EDITOR
+    UFUNCTION( BlueprintCallable )
+    void SaveSplineMarkers( TArray< FSFSplineMarker > markers_to_save );
```
Comment (Frickerson) [inline Source/SplineFollow/Public/Components/SFSplineComponent.h]: TArray &

File: Source/SplineFollow/Private/Components/SFSplineComponent.cpp [hunk 2]
```diff
@@ -128,6 +128,41 @@ void USFSplineComponent::Serialize( FArchive & archive )
 }
 
 #if WITH_EDITOR
+void USFSplineComponent::SaveSplineMarkers( TArray< FSFSplineMarker > markers_to_save )
+{
+    SplineMarkers.Reset();
+
+    const auto create_marker = [ & ]< typename U >( const auto & marker, U * object ) {
```
Comment (Frickerson) [inline Source/SplineFollow/Private/Components/SFSplineComponent.cpp]: I think you can remove the typename and just use UObject instead

**Review by Frickerson — CHANGES_REQUESTED** [review]

File: Source/SplineFollow/Public/Components/SFSplineComponent.h (hunk 1 above)
Comment (Fredaly) [reply Source/SplineFollow/Public/Components/SFSplineComponent.h]: Done

File: Source/SplineFollow/Private/Components/SFSplineComponent.cpp (hunk 2 above)
Comment (Fredaly) [reply Source/SplineFollow/Private/Components/SFSplineComponent.cpp]: done

**Review by Frickerson — APPROVED** [review]

**Review by TheEmidee — APPROVED** [review]

**Pull request merged**